File: 235.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    private TreeNode findNode = null;
    private int accCredit(TreeNode node, int val1, int val2) {
        if (node == null) {
            return 0;
        }
        int val = 0;
        if (node.val == val1) {
            val += 1;
        }
        if (node.val == val2) {
            val += 1;
        }
        val += accCredit(node.left, val1, val2);
        val += accCredit(node.right, val1, val2);
        if (val == 2 && findNode == null) {
            this.findNode = node;
        }
        return val;
    }
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        this.findNode = null;
        accCredit(root, p.val, q.val);
        return this.findNode;
    }
}

File: 235.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def accCredit(self, node, p, q):
        if not node:
            return 0, None
        val = 0
        val += 1 if node is p or node is q else 0
        c, n = self.accCredit(node.left, p, q)
        if n:
            return 0, n
        val += c
        c, n2 += self.accCredit(node.right, p, q)
        if n2:
            return 0, n2
        val += c
        return val, None
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        _, node = self.accCredit(root, p, q)
        return node